cmd/k8s-operator/svc.go

```diff
@@ -122,24 +122,34 @@ func (a *ServiceReconciler) maybeProvision(ctx context.Context, logger *zap.Suga
 		tags = strings.Split(tstr, ",")
 	}
 
-	clusterIPAddr, err := netip.ParseAddr(svc.Spec.ClusterIP)
-	if err != nil {
-		return fmt.Errorf("failed to parse cluster IP: %w", err)
-	}
-
 	sts := &tailscaleSTSConfig{
 		ParentResourceName:  svc.Name,
 		ParentResourceUID:   string(svc.UID),
-		TargetIP:            svc.Spec.ClusterIP,
+		ClusterTargetIP:     svc.Spec.ClusterIP,
```

Comment:
hmm, i think we want to not set this when `TailnetTargetIP` is also set.
We probably want to prefer `AnnotationTailnetTargetIP`, or even bail early/raise events if `AnnotationTailnetTargetIP` is set and there is a selector defined.

Can fix in a follow up though.

Reply:
https://github.com/tailscale/tailscale/pull/9155

Reply:
I've done the easiest thing in https://github.com/tailscale/tailscale/pull/9155

We could exit early in a reconciler if we see a Service that is invalid, but I am not sure about throwing an event against the Service- that will again trigger our own reconciler and we'll be just spamming kube apiserver and blocking our own reconcile loop. Maybe we could set our condition on the Service status- I'll try to do that in a separate PR

Reply:
@maisem we could set our own status condition on Services like so https://github.com/tailscale/tailscale/commit/fc7267fddb113ecfc9efb9f81af69b7a7aedcb41

```
apiVersion: v1
kind: Service
metadata:
  annotations:
    tailscale.com/expose: "true"
    tailscale.com/ts-tailnet-target-ip: 100.68.29.93
  finalizers:
  - tailscale.com/finalizer
  name: invalid
  namespace: default
spec:
  externalName: placeholder
  type: ExternalName
status:
  conditions:
  - lastTransitionTime: "2023-08-30T10:50:43Z"
    message: Service has both tailscale.com/tailnet-target-ip and tailscale.com/expose
      annotation set.
    reason: InvalidTailscaleService
    status: "False"
    type: TailscaleStatus
```